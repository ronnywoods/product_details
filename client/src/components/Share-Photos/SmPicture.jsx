import React from 'react';
import PropTypes from 'prop-types';

const SmPicture = props => {
  return (
    <li>
      <div className="overlay-blk" onMouseEnter={props.enter} onMouseLeave={props.leave}>
        <img className="sm-share img-fluid" src={props.img}></img>
      </div>
      <span className={props.class}>{props.user}</span>
    </li>
  );
};

SmPicture.propTypes = {
  enter: PropTypes.func,
  leave: PropTypes.func,
  user: PropTypes.string,
  class: PropTypes.string,
  img: PropTypes.string,
}

export default SmPicture;
