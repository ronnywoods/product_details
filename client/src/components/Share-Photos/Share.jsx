import React, { Component } from 'react';
import SmPicture from './SmPicture';

class Share extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user1: false,
      user2: false,
      user3: false,
      user4: false,
      user5: false,
    };

    this.handleEnter1 = this.handleEnter1.bind(this);
    this.handleLeave1 = this.handleLeave1.bind(this);
    this.handleEnter2 = this.handleEnter2.bind(this);
    this.handleLeave2 = this.handleLeave2.bind(this);
    this.handleEnter3 = this.handleEnter3.bind(this);
    this.handleLeave3 = this.handleLeave3.bind(this);
    this.handleEnter4 = this.handleEnter4.bind(this);
    this.handleLeave4 = this.handleLeave4.bind(this);
    this.handleEnter5 = this.handleEnter5.bind(this);
    this.handleLeave5 = this.handleLeave5.bind(this);
  }

  handleEnter1() {
    this.setState({ user1: true });
  }

  handleLeave1() {
    this.setState({ user1: false });
  }

  handleEnter2() {
    this.setState({ user2: true });
  }

  handleLeave2() {
    this.setState({ user2: false });
  }

  handleEnter3() {
    this.setState({ user3: true });
  }

  handleLeave3() {
    this.setState({ user3: false });
  }

  handleEnter4() {
    this.setState({ user4: true });
  }

  handleLeave4() {
    this.setState({ user4: false });
  }

  handleEnter5() {
    this.setState({ user5: true });
  }

  handleLeave5() {
    this.setState({ user5: false });
  }

  render() {
    const class1 = this.state.user1 ? 'share-username' : 'share-username hidden';
    const class2 = this.state.user2 ? 'share-username' : 'share-username hidden';
    const class3 = this.state.user3 ? 'share-username' : 'share-username hidden';
    const class4 = this.state.user4 ? 'share-username' : 'share-username hidden';
    const class5 = this.state.user5 ? 'share-username' : 'share-username hidden';

    return (
      <div className="share">
        <div className="title-1000">
          <h1> SHARE HOW YOU WEAR IT </h1>
          <p className="share-desc">
          Share a photo of your adidas favorite and appear in our showcase
          below. Make sure to tag your image with @adidas
          </p>
        </div>
        <div className="col-2-12 justify-content-center share-photo">
          <ul>
            <li>
              <div className="overlay-blk" onMouseEnter={this.handleEnter1} onMouseLeave={this.handleLeave1}>
                <img className="lrg-share img-fluid" src={this.props.shares.img1} />
              </div>
              <span className={class1}>{this.props.shares.user1}</span>
            </li>
          </ul>
          <ul>
            <SmPicture
              enter={this.handleEnter2}
              leave={this.handleLeave2}
              user={this.props.shares.user2}
              img={this.props.shares.img2}
              class={class2}
            />
            <SmPicture
              enter={this.handleEnter3}
              leave={this.handleLeave3}
              user={this.props.shares.user3}
              img={this.props.shares.img3}
              class={class3}
            />
          </ul>
          <ul>
            <SmPicture
              enter={this.handleEnter4}
              leave={this.handleLeave4}
              user={this.props.shares.user4}
              img={this.props.shares.img4}
              class={class4}
            />
            <SmPicture
              enter={this.handleEnter5}
              leave={this.handleLeave5}
              user={this.props.shares.user5}
              img={this.props.shares.img5}
              class={class5}
            />
          </ul>
        </div>
      </div>
    );
  }
}

export default Share;
