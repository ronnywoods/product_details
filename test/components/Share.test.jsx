import React from 'react';
import { shallow } from 'enzyme';

import Share from '../../client/src/components/Share-Photos/Share';
import SmPicture from '../../client/src/components/Share-Photos/SmPicture';

const sample = {
  id: 1,
  user1: 'anonymouse',
  img1: 'https://s3-us-west-1.amazonaws.com/adidas-shoe/share3.jpg',
  user2: 'indigosparkle',
  img2: 'https://s3-us-west-1.amazonaws.com/adidas-shoe/share7.jpg',
  user3: 'weareliving',
  img3: 'https://s3-us-west-1.amazonaws.com/adidas-shoe/share1.jpg',
  user4: 'urk3wl',
  img4: 'https://s3-us-west-1.amazonaws.com/adidas-shoe/share5.jpg',
  user5: 'cereal_killer',
  img5: 'https://s3-us-west-1.amazonaws.com/adidas-shoe/share3.jpg',
};

describe('Share Component', () => {
  const wrapper = shallow(<Share shares={sample} />);
  const picture = shallow(<SmPicture user={sample.user5} class={'share-username hidden'} />);

  it('should look like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have 1 Large picture', () => {
    expect(wrapper.find('div.overlay-blk')).toHaveLength(1);
  });

  it('should call Small Picture component 4 times', () => {
    expect(wrapper.find(SmPicture)).toHaveLength(4);
  })

  it('should not show username on normal display', () => {
    expect(wrapper.find('span').at(0)).toHaveClassName('hidden');
    expect(picture.find('span')).toHaveClassName('hidden');
  });

  it('should display username on hover/mouseenter', () => {
    wrapper.find('div.overlay-blk').at(0).prop('onMouseEnter')();
    expect(wrapper.find('span').at(0).hasClass('hidden')).toBe(false);
  });

  it('should display correct username on hover', () => {
    wrapper.find('div.overlay-blk').at(0).prop('onMouseEnter')();
    expect(wrapper.find('span').at(0).hasClass('hidden')).toBe(false);
    expect(wrapper.find('span').at(0).text()).toBe('anonymouse');
  });

  it('should not display username on mouse leave', () => {
    wrapper.find('div.overlay-blk').at(0).prop('onMouseEnter')();
    expect(wrapper.find('span').at(0).hasClass('hidden')).toBe(false);

    wrapper.find('div.overlay-blk').at(0).prop('onMouseLeave')();
    expect(wrapper.find('span').at(0)).toHaveClassName('hidden');
  });
});
