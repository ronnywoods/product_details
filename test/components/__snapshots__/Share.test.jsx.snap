// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share Component should look like snapshot 1`] = `
<div
  className="share"
>
  <div
    className="title-1000"
  >
    <h1>
       SHARE HOW YOU WEAR IT 
    </h1>
    <p
      className="share-desc"
    >
      Share a photo of your adidas favorite and appear in our showcase below. Make sure to tag your image with @adidas
    </p>
  </div>
  <div
    className="col-2-12 justify-content-center share-photo"
  >
    <ul>
      <li>
        <div
          className="overlay-blk"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <img
            className="lrg-share img-fluid"
            src="https://s3-us-west-1.amazonaws.com/adidas-shoe/share3.jpg"
          />
        </div>
        <span
          className="share-username hidden"
        >
          anonymouse
        </span>
      </li>
    </ul>
    <ul>
      <SmPicture
        class="share-username hidden"
        enter={[Function]}
        img="https://s3-us-west-1.amazonaws.com/adidas-shoe/share7.jpg"
        leave={[Function]}
        user="indigosparkle"
      />
      <SmPicture
        class="share-username hidden"
        enter={[Function]}
        img="https://s3-us-west-1.amazonaws.com/adidas-shoe/share1.jpg"
        leave={[Function]}
        user="weareliving"
      />
    </ul>
    <ul>
      <SmPicture
        class="share-username hidden"
        enter={[Function]}
        img="https://s3-us-west-1.amazonaws.com/adidas-shoe/share5.jpg"
        leave={[Function]}
        user="urk3wl"
      />
      <SmPicture
        class="share-username hidden"
        enter={[Function]}
        img="https://s3-us-west-1.amazonaws.com/adidas-shoe/share3.jpg"
        leave={[Function]}
        user="cereal_killer"
      />
    </ul>
  </div>
</div>
`;
